### PR TITLE
Batched insertion in table for speed up

### PR DESCRIPTION
**Pull request by akash17mittal**
Before Batching:
old_aidb_non_batched

After Batching:
aidb after batching

### CODE REVIEW

File: aidb/inference/bound_inference_service.py [hunk 1]
```diff
@@ -145,12 +145,24 @@ def get_tables(self, columns: List[str]) -> List[str]:
...
-    input_dic = {}
-    for col in self.binding.input_columns:
-      input_dic[self.convert_normalized_col_name_to_cache_col_name(col)] = pandas_dtype_to_native_type(getattr(row, col))
-    insert = self.get_insert()(self._cache_table).values(**input_dic)
-    await conn.execute(insert)
+  async def _insert_in_cache_table(self, inp_rows: List[pd.Series], conn):
+    if len(inp_rows) > 0:
+      inp_rows_df = pd.DataFrame(inp_rows).reset_index(drop=True)
+      inp_rows_df.columns = [self.convert_normalized_col_name_to_cache_col_name(col) for col in inp_rows_df.columns]
+      inp_rows_df = inp_rows_df.astype('object')
```
Comment (ddkang) [inline aidb/inference/bound_inference_service.py:152]: Do we need to cast? Ditto below

File: aidb/inference/bound_inference_service.py [hunk 2]
```diff
@@ -189,63 +201,21 @@ async def infer(self, inputs: pd.DataFrame):
...
-                    inference_results[col] = inp_row[c]
-                    break
-
-            tables = self.get_tables(self.binding.output_columns)
-            for table in tables:
-              columns = [col for col in self.binding.output_columns if col.startswith(table + '.')]
-              tmp_df = inference_results[columns]
-              tmp_df = tmp_df.astype('object')
-
-              if self._dialect == 'mysql' or self._dialect == 'postgresql':
```
Comment (ddkang) [inline aidb/inference/bound_inference_service.py:212]: Is it safe to delete the logic here?

File: aidb/inference/bound_inference_service.py (hunk 1 above)
Comment (akash17mittal) [reply aidb/inference/bound_inference_service.py:152]: Yes, we need this. It converts the data to python native types from pandas/numpy types. For e.g. np.int64 to int

File: aidb/inference/bound_inference_service.py (hunk 2 above)
Comment (akash17mittal) [reply aidb/inference/bound_inference_service.py:212]: I think so.

Pandas dataframe .to_sql function internally handles all the databases supported by sqlalchemy.

https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_sql.html#r689dfd12abe5-1

File: aidb/inference/bound_inference_service.py (hunk 1 above)
Comment (ddkang) [reply aidb/inference/bound_inference_service.py:152]: Can you leave a comment?

File: aidb/inference/bound_inference_service.py (hunk 2 above)
Comment (ddkang) [reply aidb/inference/bound_inference_service.py:212]: Does it deal with the upserts and stuff?

Comment (akash17mittal) [reply aidb/inference/bound_inference_service.py:212]: Nope, it doesn't support upsert. But according to our logic, this case never happens as we have cache

Comment (ddkang) [reply aidb/inference/bound_inference_service.py:212]: Can you leave a comment here too?

**Comment by ddkang** [comment]
Overall looks good to me, please review @continue-revolution @ttt-77

**Comment by continue-revolution** [comment]
also looks good to me

File: aidb/inference/bound_inference_service.py [hunk 3]
```diff
@@ -189,63 +205,21 @@ async def infer(self, inputs: pd.DataFrame):
           results.append(df)
```
Comment (ttt-77) [inline aidb/inference/bound_inference_service.py:205]: Is it also possible to perform batch selection from the cache table? If so, we can create an issue to address this at a later time.

**Comment by ttt-77** [comment]
Looks good to me!

File: aidb/inference/bound_inference_service.py (hunk 3 above)
Comment (akash17mittal) [reply aidb/inference/bound_inference_service.py:205]: It is possible. I'll handle it in different PR.

**Review by ddkang — APPROVED** [review]

**Pull request merged**